hooks/pre-command

```diff
@@ -25,17 +25,52 @@ upsert_ecr() {
   fi
 }
 
+read_build_args() {
+  if read_list_property 'BUILD_ARGS'; then
+    for arg in "${result[@]}"; do
+      build_args+=('--build-arg' "${arg}")
+    done
+  fi
+}
+
+# read a plugin property of type [array, string] into a Bash array. Buildkite
+# exposes a string value at BUILDKITE_PLUGIN_{NAME}_{KEY}, and array values at
+# BUILDKITE_PLUGIN_{NAME}_{KEY}_{IDX}.
+read_list_property() {
```

Comment:
Is this simpler than scanning over env w/prefixes? Given regex is a nice way to filter them out?

I ask as most other buildkite plugins stick to that